odl/operator/pspace_ops.py

```diff
@@ -562,6 +606,249 @@ def adjoint(self):
+                         self.prod_op.is_linear)
+
+    @property
+    def prod_op(self):
+        """The prod-op implementation"""
+        return self._prod_op
+
+    @property
+    def operators(self):
+        """A tuple of sub-operators"""
+        return self._operators
+
+    def _call(self, x, out=None):
+        """Apply operators to ``x`` and sum.
+
+        Parameters
+        ----------
+        x : domain element
+            Input vector to be extended
+        out : range element, optional
+            output vector to write result to
+
+
+        Parameters
+        ----------
+        x : domain element
+            Input vector to be evaluated by operators
+        out : range element, optional
+            output vector to write result to
+
+        Returns
+        -------
+        out : range element
+            Sum of operators evaluated in point
+
+        Examples
+        --------
+        >>> import odl
+        >>> I = odl.IdentityOperator(odl.Rn(3))
+        >>> op = ReductionOperator(I, 2 * I)
+        >>> op([[1.0, 2.0, 3.0], [4.0, 6.0, 8.0]])
+        Rn(3).element([9.0, 14.0, 19.0])
+        """
+        if out is None:
+            return self.prod_op(x)[0]
+        else:
+            wrapped_out = self.prod_op.range.element([out], cast=False)
+            return self.prod_op(x, out=wrapped_out)
+
+    def derivative(self, x):
+        """Derivative of the reduction operator.
+
+        Parameters
+        ----------
+        x : domain element
+            The point to take the derivative in
+
+        Returns
+        -------
+        adjoint : linear `BroadcastOperator`
+            The derivative
+
+        Examples
+        --------
+        >>> import odl
+        >>> I = odl.IdentityOperator(odl.Rn(3))
+        >>> x = [1.0, 2.0, 3.0]
+        >>> y = [4.0, 6.0, 8.0]
+
+        Example with linear operator (derivative is itself)
+
+        >>> op = ReductionOperator(I, 2 * I)
+        >>> op([x, y])
+        Rn(3).element([9.0, 14.0, 19.0])
+        >>> op.derivative([x, y])([x, y])
+        Rn(3).element([9.0, 14.0, 19.0])
+        """
+        return ReductionOperator(*[op.derivative(xi)
+                                   for op, xi in zip(self.operators, x)])
+
+    @property
+    def adjoint(self):
+        """Adjoint of the reduction operator.
+
+        Parameters
+        ----------
+        x : domain element
+            The point to take the derivative in
+
+        Returns
+        -------
+        adjoint : linear `BroadcastOperator`
+            The adjoint
+
+        Examples
+        --------
+        >>> import odl
+        >>> I = odl.IdentityOperator(odl.Rn(3))
+        >>> op = ReductionOperator(I, 2 * I)
+        >>> op.adjoint([1, 2, 3])
+        ProductSpace(Rn(3), 2).element([
+            [1.0, 2.0, 3.0],
+            [2.0, 4.0, 6.0]
+        ])
+        """
+        return BroadcastOperator(*[op.adjoint for op in self.operators])
+
+
```

Comment:
All good I'd say, apart from some doc stuff. The weighting issue lurks also here, but an upstream fix to that will fix this, too.